packages/geo/src/lib/layer/layer-list/layer-list.component.ts

```diff
@@ -604,6 +607,17 @@ export class LayerListComponent implements OnInit, OnDestroy {
     return;
   }
 
+  toggleVisibility( layers?: Layer[] ) {
+    if (layers && layers.length > 0) {
+      for (const layer of layers) {
+			layer.visible = this.hideSelectedLayers ;
+      }
+	  
+	}
+	  this.hideSelectedLayers = ! this.hideSelectedLayers ;
+    return;
```

Comment:
ajouté une détection de changement à la fin de la fonction:
https://angular.io/api/core/ChangeDetectorRef#detach-change-detector-to-limit-how-often-check-occurs

Reply:
Je vais avoir besoin de plus de temps pour analyser et comprendre le lien en référence.

Reply:
Bonjour

Je ne sais pas comment faire interagir le layer-list-tool avec le layerlistComponent.

SPÉCIFICATION
**En mode normal (pas seulement en mode sélection multiple), ajouter le même bouton qui servirait à ouvrir ou fermer toutes les couches du contexte.**

Je crois aussi que j'ai besoin d'aide pour la détection de changement.
En ce moment, le constructeur de layer-list.component.ts est le suivant :

  constructor(
    private elRef: ElementRef,
  ) {}

Donc j'ai essayé
- - - -
  private ref: ElementRef;
  
  constructor(
    private elRef: ElementRef,
  ) { this.ref = elRef;}
- - - - - 
	  this.ref.detectChanges();
- - - - - 
Mais j'ai une erreur  : 
>> Property 'detectChanges' does not exist on type 'ElementRef<any>'.

Donc, je suis à une impasse

Reply:
Bonjour

J'ai fait le maximum de modifications demandées. J'ai poussé les modifications dans ma brnache.
 >git push origin oeil-layer-list
La commande s'est terminée sans erreur.

Je n'ai pas tout réussi, mais je n'ai plus tellement de temps à consacrer à cette modification.
L'expérience a été très enrichissante. Je ne peux pas dire que je comprends tout le projet igo, mais je suis fier d'avoir ajouté ma modeste contribution à ce projet.

Cordialement,

Eric Darras

Reply:
Merci pour ta contribution!